query/src/main/java/org/infinispan/query/dsl/embedded/impl/EmbeddedLuceneQuery.java

```diff
@@ -76,13 +74,24 @@ public void resetQuery() {
 
    @Override
    @SuppressWarnings("unchecked")
-   public <T> List<T> list() {
-      if (results == null) {
-         List<?> list = createCacheQuery().list();
-         results = rowProcessor == null ? list : ((List<Object[]>) list).stream().map(rowProcessor)
-               .collect(Collectors.toCollection(() -> new ArrayList<>(list.size())));
-      }
-      return (List<T>) results;
+   public List<T> list() {
+      return execute().list();
+   }
+
+   @Override
+   public QueryResult<T> execute() {
+      IndexedQuery<T> cacheQuery = createCacheQuery();
+      List<T> list = cacheQuery.list();
+      int hits = cacheQuery.getResultSize();
+      List<?> results = rowProcessor == null ? list : ((List<Object[]>) list).stream().map(rowProcessor)
+            .collect(Collectors.toCollection(() -> new ArrayList<>(list.size())));
+
+      return new QueryResultImpl<>(OptionalLong.of(hits), (List<T>) results);
+   }
+
+   @Override
+   public CloseableIterator<T> iterator() {
+      return createCacheQuery().iterator();
```

Comment:
I see this method does something similar to ```execute``` minus the optional processing with the ```rowProcessor```

Reply:
right, this should have caused a test failure somewhere, unless there is no test with iterator + projections, I will check...

Reply:
"iterator + projections + indexed query" is tested and it works. There should be a hidden case somewhere I am not seeing, maybe it's related to this comment 
```// but some projections are duplicated and Hibernate Serach does not allow duplicate projections ...```

Reply:
It seems that in embedded this is not an issue since the rowProcessor is always null:
https://github.com/infinispan/infinispan/blob/a922cc2104ead99670614cae441937d35e04a9ce/query/src/main/java/org/infinispan/query/dsl/embedded/impl/QueryEngine.java#L659-L661

This ```rowProcessor``` is only used for remote query, but iterator is not currently supported for ```RemoteQuery```

Reply:
I will fix this nonetheless

Reply:
fixed